src/app/clusters/user-label-server/user-label-server.cpp

```diff
@@ -74,7 +78,27 @@ CHIP_ERROR UserLabelAttrAccess::ReadLabelList(EndpointId endpoint, AttributeValu
     return err;
 }
 
-UserLabelAttrAccess gAttrAccess;
+CHIP_ERROR UserLabelAttrAccess::WriteLabelList(EndpointId endpoint, AttributeValueDecoder & aDecoder)
+{
+    DeviceLayer::LabelList<app::Clusters::UserLabel::Structs::LabelStruct::Type, DeviceLayer::kMaxUserLabels> labelList;
+    LabelList::TypeInfo::DecodableType decodablelist;
+
+    ReturnErrorOnFailure(aDecoder.Decode(decodablelist));
+
+    uint8_t index = 0;
+    auto iter     = decodablelist.begin();
+    while (iter.Next())
+    {
+        auto & entry = iter.GetValue();
+
+        VerifyOrReturnError(index < DeviceLayer::kMaxUserLabels, CHIP_ERROR_BUFFER_TOO_SMALL);
+        labelList.add(entry);
+    }
+
+    ReturnErrorOnFailure(DeviceLayer::PlatformMgr().SetUserLabelList(endpoint, labelList));
```

Comment:
Why are we doing this before checking `iter.GetStatus()`?  If the incoming value is not actually decodable as a valid list, do we want to overwrite the stored value with an empty list?  This does not seem right.